test/extended/util/framework.go

```diff
@@ -265,6 +300,79 @@ func CreatePodForImage(dockerImageReference string) *kapi.Pod {
 	}
 }
 
+// CreatePersistentVolume creates a HostPath Persistent Volume.
+func CreatePersistentVolume(name, capacity, hostPath string) *kapi.PersistentVolume {
+	return &kapi.PersistentVolume{
+		TypeMeta: kapi.TypeMeta{
+			Kind:       "PersistentVolume",
+			APIVersion: "v1",
+		},
+		ObjectMeta: kapi.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"name": name},
+		},
+		Spec: kapi.PersistentVolumeSpec{
+			PersistentVolumeSource: kapi.PersistentVolumeSource{
+				HostPath: &kapi.HostPathVolumeSource{
+					Path: hostPath,
+				},
+			},
+			Capacity: kapi.ResourceList{
+				kapi.ResourceStorage: resource.MustParse(capacity),
+			},
+			AccessModes: []kapi.PersistentVolumeAccessMode{
+				kapi.ReadWriteOnce,
+				// TODO: Uncomment after https://github.com/kubernetes/kubernetes/pull/10833 is merged
+				//kapi.ReadOnlyMany,
+				//kapi.ReadWriteMany,
+			},
+		},
+	}
+}
+
+// SetupHostPathVolumes will create multiple PersistentVolumes with given capacity
+func SetupHostPathVolumes(c kclient.PersistentVolumeInterface, prefix, capacity string, count int) (volumes []*kapi.PersistentVolume, err error) {
+	if count < 1 {
+		return volumes, fmt.Errorf("Invalid number of volumes: '%v'", count)
+	}
+
+	rootDir, err := ioutil.TempDir(TestContext.OutputDir, "persistent-volumes")
+	if err != nil {
+		return volumes, err
+	}
+	for i := 0; i < count; i++ {
+		dir, err := ioutil.TempDir(rootDir, fmt.Sprintf("%0.4d", i))
```

Comment:
yup, that's what we do in our own e2e https://github.com/openshift/origin/blob/master/hack/test-end-to-end-docker.sh#L28